### ISSUE TITLE
[amp-story] Videos not working in stories

### ISSUE THREAD

**Issue by Enriqe** [ampproject/amphtml#32323]
On AMP version 2101230412002
MacOS Safari 14.0.2

Go to https://www.gstatic.com/amphtml/stamp/qa/navigation.html

Video on page 7 will get stuck ~60% of the time when navigating to it. 

https://user-images.githubusercontent.com/5449100/106650529-04b2ec80-6561-11eb-8839-0de97a2f361b.mov

Checking the video element, it seems like `play()` was never called on the element, as it is still has `video.paused === true`.

Also happens in https://www.gstatic.com/amphtml/stamp/qa/video.html

In any other platform (e.g. Chrome in MacOS) go to the next URL https://www.vice.com/stories/making-20k-a-month-selling-street-cocktails/

After the 8th video, no videos will play and the "Video failed to play" message will show up.

/cc @ampproject/wg-stories

**Comment by processprocess** [comment]
I'm not able to repro. 
Seems to always play on direct navigation and auto-advance.

Does the network request complete? 
I wonder if the weather might be effecting connection speed.

**Comment by Enriqe** [comment]
Could you double check you're testing the right AMP version and with Safari version? I'm able to consistently repro with those.


The fact that calling `element.play()` on the video element and that `element.paused` suggests that it is not an issue with the network, but either with a) our runtime is not calling `play()` at the right time or b) a change in Safari is causing this issue.

**Comment by newmuis** [comment]
I can repro, on Mac OS Chrome.  Let's treat this as P0, maybe blocking the `2101230412002` release

**Comment by newmuis** [comment]
Seems to be working for me in `2101211748002`.

The only differences are https://github.com/ampproject/amphtml/compare/2101230412002...2101211748002

**Comment by Enriqe** [comment]
We found the potential culprit, which comes from https://github.com/ampproject/amphtml/pull/31976

`amp-video` introduces a `detachedCallback()` which updates the video state to false.

In stories the media pool recycles videos (detaches and re-attaches same video elements in the DOM), which means that whenever a video is detached from the DOM it will call `unobserveDisplay()`. Not sure yet what this does, maybe @dvoytenko can provide more details.

**Comment by Enriqe** [comment]
To clarify, I think we found 2 different bugs here: 

1) is described by the first comment in this issue
2) when a story has more than 8 videos, the next videos will not play. This is reproducible in MacOS Chrome

**Comment by gmajoulet** [comment]
What would a patch look like?
Can we get away with a couple `if (managedByPool) return`?

**Comment by Enriqe** [comment]
I've created a draft PR with the necessary changes. I think we could get away with the `if (managedByPool) return`

https://github.com/ampproject/amphtml/pull/32355

**Comment by Enriqe** [comment]
Confirmed that by short-circuiting the 

```  
updateIsPlaying_(isPlaying) {
    if (this.isManagedByPool_()) {
      return;
    }
```

in amp-video, the videos start to work again for stories.

**Comment by Enriqe** [comment]
Closed by #32355

**Comment by newmuis** [comment]
This is still an issue in `2101300534004`.
